accelerator/x86_accelerator.py

```diff
@@ -0,0 +1,297 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import os
+import pkgutil
+import importlib
+
+from .abstract_accelerator import DeepSpeedAccelerator
+# During setup stage torch may not be installed, pass on no torch will
+# allow op builder related API to be executed.
+try:
+    import torch.cpu
+except ImportError:
+    pass
+
+
+class x86_Accelerator(DeepSpeedAccelerator):
+
+    def __init__(self):
+        self._name = 'x86'
+
+    def is_synchronized_device(self):
+        return False
+
+    # Device APIs
+    def device_name(self, device_index=None):
+        if device_index is None:
+            return 'x86'
+        return 'x86:{}'.format(device_index)
```

Comment:
Are we sure this is naming convention for torch on cpu?

Reply:
It isn't, the naming convention is `cpu`, so this needs to be updated.  The first pass was largely find/replace, though we will also need to figure out how to handle this with the current cpu_accelerator.

Reply:
Sounds good to me. Actually, based on my chat with Mike, this file should mostly mirror cpu_accelerator while cpu_accelerator is renamed to xeon_accelerator. I am happy to chat more about this.

Reply:
Hi @tjruwase, What is the hurdle that we can't merge x86 and xeon accelerator? We could try to make it work for all x86, given xeon is x86.

Reply:
Agreed, I've renamed the accelerator, though the git tracking of this treats it as a new file, so the review will be somewhat difficult.  If we want to rename the cpu_accelerator -> xeon_accelerator in a PR first that wil be easier for git to track.  Thoughts?

Reply:
I created #5126 to better preserve git commit history